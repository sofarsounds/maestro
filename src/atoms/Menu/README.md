# Menu

To implement a `Menu` wrapper into your project you'll need to add this import
```js
import { Menu } from '@sofarsounds/maestro'
```

After adding the import you can use any button simply like this
```html
<Menu>
  Content in my menu
</Menu>
```

## Props
Table below contains all types of props available in the Menu component  

| Name          | Type        | Default         | Description                      |
| :------------ | :-----      | :-------------- | :------------------------------- |
| *children*    | `ReactNode` |                 | The content to be displayed
| bordered      | `Boolean`   | `false`         | Whether to add a border around the menu
| width         | `string `   | `false`         | Optional width for the logo
| depth         | `number`    | `3`             | Shadow depth
| data-qaid     | `string`    |                 | Optional prop for testing purposes



