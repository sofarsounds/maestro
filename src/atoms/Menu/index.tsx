import styled, { css } from '../../lib/styledComponents';
import { withShadow } from '../../util';
import { ShadowDepth } from '../../util/withShadow';

interface Props {
  children?: any;
  width?: string;
  bordered?: boolean;
  depth?: any;
}

const Menu = styled.ul<Props>`
  ${({ depth = 3, theme, width, bordered }) => css`
    ${withShadow({ depth })};

    margin: 0;
    padding: 0;
    border-radius: ${theme.borderRadius.default};
    background-color: ${theme.colors.whiteDenim};
    max-height: 300px;
    width: ${width || 'auto'};
    overflow-y: auto;
    -webkit-overflow-scrolling: touch;

    ${bordered &&
      css`
        border: 1px solid ${theme.colors.macyGrey};
      `}
  `};
`;

export default Menu;
