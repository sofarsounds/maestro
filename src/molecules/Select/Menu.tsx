/* import React from 'react'; */
import styled, { css } from '../../lib/styledComponents';
import { withShadow } from '../../util';

interface Props {
  children?: any;
  depth: any;
  width?: number;
  isSelect?: boolean;
}

const Menu = styled.ul<Props>`
  ${({ theme, width, isSelect = false }) => css`
    ${withShadow}
    margin-top: 0px;
    border-bottom-left-radius: ${theme.borderRadius.large};
    border-bottom-right-radius: ${theme.borderRadius.large};
    width: ${width}px;
    padding-left: 0px;
    background-color: ${theme.colors.whiteDenim};
    max-height: 300px;
    overflow-y: auto;

    ${!width &&
      css`
        width: auto;
      `}

    ${width &&
      css`
        width: ${width}px;
      `}

    ${!isSelect &&
      css`
        border-top-left-radius: ${theme.borderRadius.large};
        border-top-right-radius: ${theme.borderRadius.large};
      `}
  `};
`;

export default Menu;
