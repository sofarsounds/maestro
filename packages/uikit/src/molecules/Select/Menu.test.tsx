import React from 'react';

import { mountWithTheme } from '../../test';
import Menu from './Menu';

const setup = () =>
  mountWithTheme(<Menu depth={2}>I am a child component</Menu>);

describe('Select <Menu />', () => {
  it('it renders correctly', () => {
    expect(setup()).toMatchSnapshot();
  });

  it('it renders child components', () => {
    expect(setup().props().children.props.children).toBe(
      'I am a child component'
    );
  });

  it('it shows the utility shadow css properties', () => {
    expect(setup()).toHaveStyleRule(
      'box-shadow',
      '0 3px 6px 0 rgba(0,0,0,0.16), 0 3px 6px 0 rgba(0,0,0,0.23)'
    );
  });
});
