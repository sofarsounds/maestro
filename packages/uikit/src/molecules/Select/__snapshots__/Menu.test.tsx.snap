// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select <Menu /> it renders correctly 1`] = `
.c0 {
  box-shadow: 0 3px 6px 0 rgba(0,0,0,0.16), 0 3px 6px 0 rgba(0,0,0,0.23);
  margin-top: 0px;
  padding-left: 0px;
  background-color: #FFFFFF;
}

<ThemeProvider
  theme={
    Object {
      "borderRadius": Object {
        "button": "4px",
        "input": "2px",
      },
      "colours": Object {
        "backToBlack": "#111111",
        "black": "#000000",
        "blackBetty": "#1E1E1E",
        "blueSmoke": "#949494",
        "green100": "#CFEACC",
        "green200": "#9DDCAC",
        "green300": "#72D08A",
        "green400": "#4FC571",
        "green50": "#E6F7EA",
        "green500": "#23BA57",
        "green600": "#10AD52",
        "green700": "#039842",
        "green800": "#008736",
        "green900": "#006721",
        "macyGrey": "#DBDBDB",
        "paintItBlack": "#333333",
        "primary": "#039842",
        "redRedWine": "#EF4423",
        "silverSprings": "#F5F5F5",
        "whiteDenim": "#FFFFFF",
      },
      "dimensions": Object {
        "buttonHeight": "44px",
        "containerWidth": Object {
          "default": "800px",
          "fullscreen": "100%",
          "large": "1000px",
          "small": "600px",
        },
        "inputHeight": "44px",
      },
      "fontSizes": Object {
        "base": "16px",
        "body1": "16px",
        "body2": "14px",
        "button": "14px",
        "caption": "10px",
        "h1": "96px",
        "h2": "60px",
        "h3": "48px",
        "h4": "34px",
        "h5": "24px",
        "h6": "20px",
        "overline": "10px",
        "subtitle1": "16px",
        "subtitle2": "14px",
        "title": "74px",
      },
      "fonts": Object {
        "regular": "Open Sans",
      },
      "media": Object {
        "lg": [Function],
        "md": [Function],
        "sm": [Function],
        "xl": [Function],
        "xs": [Function],
      },
      "ruler": Array [
        0,
        4,
        8,
        12,
        16,
        20,
        24,
        28,
        32,
        36,
        40,
        44,
        48,
        52,
        56,
        60,
        64,
        68,
        72,
        76,
        80,
      ],
      "utils": Object {
        "transition": [Function],
      },
    }
  }
>
  <styled.ul
    depth={2}
  >
    <StyledComponent
      depth={2}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              [Function],
              ";
",
            ],
          },
          "displayName": "styled.ul",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "ul",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <ul
        className="c0"
      >
        I am a child component
      </ul>
    </StyledComponent>
  </styled.ul>
</ThemeProvider>
`;

exports[`Select <Menu /> it renders correctly when the first page is active 1`] = `
.c0 {
  box-shadow: 0 3px 6px 0 rgba(0,0,0,0.16), 0 3px 6px 0 rgba(0,0,0,0.23);
  margin-top: 0px;
  padding-left: 0px;
  background-color: #FFFFFF;
}

<ThemeProvider
  theme={
    Object {
      "borderRadius": Object {
        "button": "4px",
        "input": "2px",
      },
      "colours": Object {
        "backToBlack": "#111111",
        "black": "#000000",
        "blackBetty": "#1E1E1E",
        "blueSmoke": "#949494",
        "green100": "#CFEACC",
        "green200": "#9DDCAC",
        "green300": "#72D08A",
        "green400": "#4FC571",
        "green50": "#E6F7EA",
        "green500": "#23BA57",
        "green600": "#10AD52",
        "green700": "#039842",
        "green800": "#008736",
        "green900": "#006721",
        "macyGrey": "#DBDBDB",
        "paintItBlack": "#333333",
        "primary": "#039842",
        "redRedWine": "#EF4423",
        "silverSprings": "#F5F5F5",
        "whiteDenim": "#FFFFFF",
      },
      "dimensions": Object {
        "buttonHeight": "44px",
        "containerWidth": Object {
          "default": "800px",
          "fullscreen": "100%",
          "large": "1000px",
          "small": "600px",
        },
        "inputHeight": "44px",
      },
      "fontSizes": Object {
        "base": "16px",
        "body1": "16px",
        "body2": "14px",
        "button": "14px",
        "caption": "10px",
        "h1": "96px",
        "h2": "60px",
        "h3": "48px",
        "h4": "34px",
        "h5": "24px",
        "h6": "20px",
        "overline": "10px",
        "subtitle1": "16px",
        "subtitle2": "14px",
        "title": "74px",
      },
      "fonts": Object {
        "regular": "Open Sans",
      },
      "media": Object {
        "lg": [Function],
        "md": [Function],
        "sm": [Function],
        "xl": [Function],
        "xs": [Function],
      },
      "ruler": Array [
        0,
        4,
        8,
        12,
        16,
        20,
        24,
        28,
        32,
        36,
        40,
        44,
        48,
        52,
        56,
        60,
        64,
        68,
        72,
        76,
        80,
      ],
      "utils": Object {
        "transition": [Function],
      },
    }
  }
>
  <styled.ul
    depth={2}
  >
    <StyledComponent
      depth={2}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              [Function],
              ";
",
            ],
          },
          "displayName": "styled.ul",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "ul",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <ul
        className="c0"
      >
        I am a child component
      </ul>
    </StyledComponent>
  </styled.ul>
</ThemeProvider>
`;
